compiler/ballerina-parser/src/main/java/io/ballerina/compiler/syntax/tree/NodeParser.java

```diff
@@ -170,4 +173,17 @@ public static LetVariableDeclarationNode parseLetVarDeclaration(String text, boo
         BallerinaParser parser = ParserFactory.getParser(text);
         return parser.parseAsLetVarDeclaration(allowActions).createUnlinkedFacade();
     }
+
+    /**
+     * Parses the input as a regexp pattern.
+     *
+     * @param text the input
+     * @return an {@code ExpressionNode}
+     */
+    public static ExpressionNode parseRegexpPattern(String text) {
+        String regexpStr = RE_KEYWORD.stringValue() + BACKTICK_TOKEN.stringValue() + text
+                            + BACKTICK_TOKEN.stringValue();
```

Comment:
This injecting part is troublesome.  For example, let's say the text is `[A-Z]` and we get the position of `]`. The correct position should be `4`. Since we inject a text wrapper inside the positions will be affected. 

Simply, input `[A-Z]` => output: ``Re `[A-Z]` `` which I think is not a suitable behavior for a public API. 

We can achieve the same output using our existing API,
 ``parseExpression(String.Format("`Re `%s`", "[A-Z]" ))``. 

Maybe we can have the above as a utility method in open API if the usages are prevalent. 

@rdulmina your thoughts on this?

Reply:
Additionally, we can't expose an API where input and output both being `[A-Z]` since we have defined the content within the backticks as a `NodeList`. (I have described the reason in #33323)
https://github.com/ballerina-platform/ballerina-lang/blob/4df9b3836b0020fab1e06136a28265152d30b4b3/compiler/ballerina-treegen/src/main/resources/syntax_tree_descriptor.json#L2143-L2147

Reply:
Yes, I think the user will have to use `parseExpression(String.Format("`Re `%s`", "[A-Z]" ))` for this use-case without a separate API.

Reply:
@lnash94 Would this do-able from the OpenAPI tooling side?

Reply:
Since we always have the `re` prefix it will always generate regular expression. If the regexp is invalid it will anyway create a valid node with diagnostic.

Reply:
you can simply use, `node.hasDiagnostics()` to see if if the regex is syntactically valid.